operators/src/source/ogr_source.rs

```diff
@@ -501,43 +546,36 @@ where
         // TODO: add OGR time filter if forced
         let dataset = Self::open_gdal_dataset(dataset_information)?;
 
-        let mut use_ogr_spatial_filter = dataset_information.force_ogr_spatial_filter;
-
         let mut features_provider = if let Some(sql) = dataset_information.sql_query.as_ref() {
-            let spatial_filter = if use_ogr_spatial_filter {
-                Some(
-                    query_rectangle
-                        .spatial_bounds
-                        .try_into()
-                        .context(error::DataType)?,
-                )
-            } else {
-                None
-            };
-
             FeaturesProvider::ResultSet(
                 dataset
-                    .execute_sql(sql, spatial_filter.as_ref(), Dialect::DEFAULT)?
+                    .execute_sql(sql, None, Dialect::DEFAULT)?
                     .ok_or(error::Error::OgrSqlQuery)?,
             )
         } else {
-            let mut layer = dataset.layer_by_name(&dataset_information.layer_name)?;
-
-            use_ogr_spatial_filter = dataset_information.force_ogr_spatial_filter
-                || layer.has_capability(gdal::vector::LayerCaps::OLCFastSpatialFilter);
-
-            if use_ogr_spatial_filter {
-                // rectangular geometry from West, South, East and North values.
-                let filter_geometry = query_rectangle.spatial_bounds.try_into()?;
-                // TODO: log uses spatial filter
-                // TODO: use OGR_L_SetSpatialFilterRect() once GDAL crate supports it
-                // NOTE: the OGR-filter may be inaccurately allowing more features that should be returned in a "strict" fashion.
-                layer.set_spatial_filter(&filter_geometry);
-            }
-
-            FeaturesProvider::Layer(layer)
+            FeaturesProvider::Layer(dataset.layer_by_name(&dataset_information.layer_name)?)
         };
 
+        let use_ogr_spatial_filter = dataset_information.force_ogr_spatial_filter
+            || features_provider.has_gdal_capability(gdal::vector::LayerCaps::OLCFastSpatialFilter);
+
+        if use_ogr_spatial_filter {
+            debug!(
+                "using spatial filter {:?} for layer {:?}",
+                query_rectangle.spatial_bounds, &dataset_information.layer_name
+            );
+            // NOTE: the OGR-filter may be inaccurately allowing more features that should be returned in a "strict" fashion.
```

Comment:
do you mean "than in a strict fashion"? Is this a problem for rectangles?

Reply:
that comment is not from me :(